services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/toggles/FeatureTogglesV2Spec.groovy

```diff
@@ -112,4 +122,159 @@ class FeatureTogglesV2Spec extends HealthCheckSpecification {
+        northbound.getFlow(flow.flowId).encapsulationType == newFlowEncapsulationType.toString().toLowerCase()
+
+        when: "Update default flow encapsulation type in kilda configuration"
+        northbound.updateKildaConfiguration(
+                new KildaConfigurationDto(flowEncapsulationType: initKildaConfig.flowEncapsulationType))
+
+        and: "Disable the 'flows_reroute_using_default_encap_type' feature toggle"
+        northbound.toggleFeature(FeatureTogglesDto.builder().flowsRerouteUsingDefaultEncapType(false).build())
+
+        and: "Restore previous path"
+        antiflap.portUp(islToBreak.srcSwitch.dpId, islToBreak.srcPort)
+        Wrappers.wait(antiflapMin + 2) {
+            assert islUtils.getIslInfo(islToBreak).get().state == IslChangeType.DISCOVERED
+        }
+
+        and: "Init a flow reroute by breaking a new current path"
+        def newCurrentPath = pathHelper.convert(northbound.getFlowPath(flow.flowId))
+        def newIslToBreak = pathHelper.getInvolvedIsls(newCurrentPath).first()
+        antiflap.portDown(newIslToBreak.srcSwitch.dpId, newIslToBreak.srcPort)
+        Wrappers.wait(antiflapMin + 2) {
+            assert islUtils.getIslInfo(islToBreak).get().state == IslChangeType.FAILED
+        }
+
+        then: "Flow is rerouted"
+        Wrappers.wait(WAIT_OFFSET + rerouteDelay) {
+            assert pathHelper.convert(northbound.getFlowPath(flow.flowId)) != newCurrentPath
+        }
+
+        and: "Encapsulation type is not changed according to kilda configuration"
+        northbound.getFlow(flow.flowId).encapsulationType != initKildaConfig.flowEncapsulationType
+
+        and: "Cleanup: Revert system to origin state"
+        northbound.toggleFeature(FeatureTogglesDto.builder()
+                .flowsRerouteUsingDefaultEncapType(initFeatureToggle.flowsRerouteUsingDefaultEncapType).build())
+        flowHelperV2.deleteFlow(flow.flowId)
+    }
+
+    @Ignore("https://github.com/telstra/open-kilda/issues/2955")
+    def "Flow encapsulation type is not changed while syncing/auto rerouting/updating according to \
+'flows_reroute_using_default_encap_type' if switch does't support new type of encapsulation"() {
+        given: "A switch which supports only transit_vlan encapsulation type"
+        def swPair = topologyHelper.getAllNeighboringSwitchPairs().find {
+            [it.src, it.dst].every { !it.features.contains(SwitchFeature.NOVIFLOW_COPY_FIELD) } && it.paths.size() >= 2
+        }
+        assumeTrue("Unable to find required switches in topology", swPair as boolean)
+
+        and: "The 'flows_reroute_using_default_encap_type' feature is enabled"
+        def initFeatureToggle = northbound.getFeatureToggles()
+        !initFeatureToggle.flowsRerouteUsingDefaultEncapType && northbound.toggleFeature(FeatureTogglesDto.builder()
+                .flowsRerouteUsingDefaultEncapType(true).build())
+
+        and: "Default encapsulation type is transit_vlan"
```

Comment:
This is not required step. You can directly create the flow as 'transit vlan' and then change default config to 'vxlan'